syncer/main.py

```diff
@@ -196,12 +501,16 @@ def get_config(self):
 
     @om_expose
     def set_config(self, config):
+        logger.info("Saving configuration")
         config = json.loads(config)
         for key in config:
             if isinstance(config[key], six.string_types):
                 config[key] = str(config[key])
         self._config_checker.check_config(config)
         self._config = config
-        self._read_config()
+        thread = Thread(target=self._read_config)
```

Comment:
Because I otherwise get startup errors as the reading of the config uses remote calls and it takes too long. It would also take double the amount of code to split the saving of the config and the processing of it. The reading of the config actually is done by "self._config = config". So if we really look into the function _read_config, this is actually a processing of the config, which includes reorganizing, adding necessary extra information, making new objects, remote calls, connectors (sensor_tdos) and doing data validation/security checks to notify the user if he did something wrong and let the plugin work with the bare minimum of information it got. So _read_config should be called _proces_config tbh.